sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -461,51 +597,301 @@ is `name`.
 
 _Throws:_
 
+* An `exception` with the `errc::invalid` error code if
+  `ext_oneapi_has_kernel(name)` returns `false`.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+std::string ext_oneapi_get_raw_kernel_name(const std::string &name)
+----
+!====
+
+_Constraints:_ This function is not available when `State` is
+`bundle_state::ext_oneapi_source`.
+
+_Returns:_ If the kernel bundle was created from a bundle of state
+`bundle_state::ext_oneapi_source` and `name` was registered via
+`registered_kernel_names`, returns the compiler-generated (e.g. mangled) name
+for this kernel function.
+If the kernel bundle was created from a bundle of state
+`bundle_state::ext_oneapi_source` and `name` is the same as a
+compiler-generated name for a kernel defined in that bundle, that same
+`name` is returned.
+
+_Throws:_
+
 * An `exception` with the `errc::invalid` error code if
   `ext_oneapi_has_kernel(name)` returns `false`.
 |====
 
+=== Obtaining a kernel when the language is `sycl`
 
-== Example
+When the kernel is defined in the language `source_language::sycl`, the host
+code may query for the kernel or obtain the `kernel` object using either the
+kernel's name as it is generated by the compiler (i.e. the C++ mangled name) or
```

Comment:
"Pro tip": Using `C++` in `asciidoc` might be problematic, as `++` are escape characters (everything between `++` and `++` won't be formatted). Add the following to the beginning of the file and use `{cpp}` instead of `C++` to avoid surprises in the future:
```
:cpp: pass:[C++]
```

I ran into some instances of `C++` rendering as `C` and it can be annoying...

Reply:
Yes.  Done in 002d47be70998e1602b156bc13296575b06083c6.